lab-anthony/app/controllers/game-controller.js

```diff
@@ -0,0 +1,114 @@
+'use strict';
+
+//npm modules
+const angular = require('angular');
+
+//app modules
+const Monster = require('../models/monster.js');
+const Gem = require('../models/gem.js');
+
+//modules logic
+angular.module('monsterMash').controller('MonsterController', [MonsterController]);
+
+function MonsterController() {
+  this.history = ['You have entered the room! Get ready to mash some monsters!'];
+  this.moveCount = 0;
+  this.map = require('../models/map');
+
+  this.gameStart = false;
+  this.gameOver = false;
+
+  this.player = {
+    name: 'Big Money',
+    hp: 10,
+    damage: 10,
+    xp: 0,
+    location: 'roomA'
+  };
+
+  this.gameForm = {
+    direction: 'east'
+  };
+
+  this.room = {
+    name: this.player.location
+  };
+}
+
+MonsterController.prototype.moveDirection = function(direction) {
+  this.isDead();
+  this.moveCount ++;
+  var oldLocation = this.player.location;
+  var newLocation = this.map[oldLocation][direction];
+  if(newLocation && newLocation !== 'wall') {
+    this.updateLocation(newLocation);
+    return;
+  }
+
+  this.holdLocation();
+};
+
+MonsterController.prototype.updateLocation = function(location) {
+  this.player.location = location;
+  this.room.name = location;
+  if (Math.random() < 0.5) {
+    this.room.gem = new Gem();
+    this.player.hp += this.room.gem.addHp;
+    this.logTurn(`you found a Gem! It boosted your hp by ${this.room.gem.addHp}`);
+  }
+  if(Math.random() < this.map[location].monsterOdds) {
+    this.room.monster = new Monster();
+    this.player.hp -= this.room.monster.damage;
+    this.logTurn(`is now in ${this.player.location}. A ${this.room.monster.name} attacked! You lost ${this.room.monster.damage}.`);
+    this.isDead();
+    return;
+  }
+
+  this.room.monster = null;
+  this.logTurn(`is now in ${this.player.location}. The room is empty.`);
+};
+
+MonsterController.prototype.holdLocation = function() {
+  this.logTurn('you hit a wall.');
+};
+
+MonsterController.prototype.logTurn = function(message) {
+  this.history.push(`Turn ${this.moveCount}: ${this.player.name} ${message}`);
+};
+
+MonsterController.prototype.hurtMonster = function() {
+  this.moveCount ++;
+  if(this.room.monster){
+    var message = '';
+    if(Math.random() > 0.4) {
+      this.player.hp -= this.room.monster.damage;
+      message += `The monster hurt you! You lost ${this.room.monster.damage}.`;
+    }
+    this.room.monster.hp -= this.player.damage;
+    if (this.room.monster.hp < 0) {
+      this.player.xp += 100;
+      this.logTurn(`You killed the ${this.room.monster.name} and gained 100 xp!`);
+      this.room.monster = null;
+      return;
+    }
+    this.logTurn(message + ' you attacked the monster.');
+  }
+  this.isDead();
+};
+
+MonsterController.prototype.gameBegin = function() {
+  this.gameStart = true;
+  this.gameOver = false;
+};
+
+MonsterController.prototype.gameEnd = function() {
+  this.gameOver = true;
+  this.history = ['You have entered the room! Get ready to mash some monsters!'];
```

Comment:
Why are you reinitializing things to their beginning state here? Don't those seem like things that would be better done in a game start function?